test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts

```diff
@@ -625,12 +630,92 @@ describe('CSOT spec prose tests', function () {
      * 1. Using `session`, execute `session.end_session`
      *    - Expect this to fail with a timeout error after no more than 15ms.
      */
+    const failpoint: FailPoint = {
+      configureFailPoint: 'failCommand',
+      mode: { times: 1 },
+      data: {
+        failCommands: ['abortTransaction'],
+        blockConnection: true,
+        blockTimeMS: 600
+      }
+    };
+
+    before('drop collection', async function () {
+      const internalClient = this.configuration.newClient();
+      await internalClient
+        .db('endSession_db')
+        .collection('endSession_coll')
+        .drop()
+        .catch(() => null);
+      await internalClient.close();
+    });
+
+    beforeEach('setup failpoint', async function () {
+      if (!semver.satisfies(this.configuration.version, '>=4.4')) {
+        this.skipReason = 'Requires server version 4.4+';
+        this.skip();
+      }
+
+      const internalClient = this.configuration.newClient();
+      await internalClient.db('admin').command(failpoint);
+      await internalClient.close();
+    });
+
+    let client: MongoClient;
+
+    afterEach(async function () {
+      if (semver.satisfies(this.configuration.version, '>=4.4')) {
+        const internalClient = this.configuration.newClient();
+        await internalClient.db('admin').command({ ...failpoint, mode: 'off' });
+        await internalClient.close();
+      }
+      await client?.close();
+    });
+
+    describe('when timeoutMS is provided to the client', () => {
+      it('throws a timeout error from endSession', metadata, async function () {
+        client = this.configuration.newClient({ timeoutMS: 500, monitorCommands: true });
+        const coll = client.db('endSession_db').collection('endSession_coll');
+        const session = client.startSession();
+        session.startTransaction();
+        await coll.insertOne({ x: 1 }, { session });
+        const error = await session.endSession().catch(error => error);
+        expect(error).to.be.instanceOf(MongoOperationTimeoutError);
```

Comment:
the test says:

> Expect this to fail with a timeout error after no more than 15ms.

Should we be asserting on the duration property you added to the timeout error?

Reply:
I no longer have access to that error, but I can track that in this test, added!